pkg/sink/pager_duty.go

```diff
@@ -0,0 +1,235 @@
+package sink
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kubeshop/botkube/internal/health"
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/sliceutil"
+)
+
+// PagerDuty provides functionality to notify PagerDuty service about new events.
+type PagerDuty struct {
+	log      logrus.FieldLogger
+	reporter AnalyticsReporter
+
+	bindings config.SinkBindings
+
+	integrationKey string
+	clusterName    string
+	pagerDutyCli   *pagerduty.Client
+
+	status        health.PlatformStatusMsg
+	failureReason health.FailureReasonMsg
+	statusMux     sync.Mutex
+}
+
+// EventLink represents a link in a ChangeEvent and Alert event
+// https://developer.pagerduty.com/docs/events-api-v2/send-change-events/#the-links-property
+type EventLink struct {
+	Href string `json:"href"`
+	Text string `json:"text,omitempty"`
+}
+
+type incomingEvent struct {
+	Source    string
+	Data      any
+	Timestamp time.Time
+}
+
+// NewPagerDuty creates a new PagerDuty instance.
+func NewPagerDuty(log logrus.FieldLogger, commGroupIdx int, c config.PagerDuty, clusterName string, reporter AnalyticsReporter) (*PagerDuty, error) {
+	var opts []pagerduty.ClientOptions
```

Comment:
AFAIR we should check here if we have the remote config enabled, right? We can also do it in main, up to you.